From 1f996ec320ac6d80300dbc3759a5d50cb1a3b8e6 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@freescale.com>
Date: Tue, 8 Sep 2015 14:21:04 +0800
Subject: [PATCH 0533/5242] MLK-11503-2 ARM: dts: imx7d-12x12-lpddr3-arm2:
 enable usdhc in lpsr mode

commit  72439bc434920fefce5fb00ed69a6eca9c613a7a from
https://source.codeaurora.org/external/imx/linux-imx.git

Add pinctrl sleep mode support for usdhc, and enable usdhc in lpsr mode.

Signed-off-by: Haibo Chen <haibo.chen@freescale.com>
(cherry picked from commit 75b33d1a578abaed44b11a05187c19dedb25aad2)

Conflicts:
	arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index 46aa5d7..75402c2 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -839,8 +839,9 @@
 };
 
 &usdhc1 {
-	pinctrl-names = "default";
+	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&pinctrl_usdhc1_1>;
+	pinctrl-1 = <&pinctrl_usdhc1_1>;
 	cd-gpios = <&gpio5 0 GPIO_ACTIVE_LOW>;
 	wp-gpios = <&gpio5 1 GPIO_ACTIVE_HIGH>;
 	no-1-8-v;
@@ -851,10 +852,11 @@
 };
 
 &usdhc2 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz", "sleep";
 	pinctrl-0 = <&pinctrl_usdhc2_1>;
 	pinctrl-1 = <&pinctrl_usdhc2_1_100mhz>;
 	pinctrl-2 = <&pinctrl_usdhc2_1_200mhz>;
+	pinctrl-3 = <&pinctrl_usdhc2_1>;
 	cd-gpios = <&gpio5 9 GPIO_ACTIVE_LOW>;
 	wp-gpios = <&gpio5 10 GPIO_ACTIVE_HIGH>;
 	keep-power-in-suspend;
@@ -865,10 +867,11 @@
 };
 
 &usdhc3 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz", "sleep";
 	pinctrl-0 = <&pinctrl_usdhc3_1>;
 	pinctrl-1 = <&pinctrl_usdhc3_1_100mhz>;
 	pinctrl-2 = <&pinctrl_usdhc3_1_200mhz>;
+	pinctrl-3 = <&pinctrl_usdhc3_1>;
 	assigned-clocks = <&clks IMX7D_USDHC3_ROOT_CLK>;
 	assigned-clock-rates = <400000000>;
 	bus-width = <8>;
-- 
1.7.9.5

